plugins/node/opentelemetry-instrumentation-winston/src/instrumentation.ts

```diff
@@ -163,40 +177,74 @@ export class WinstonInstrumentation extends InstrumentationBase {
     return (original: Winston2LogMethod) => {
       const instrumentation = this;
       return function patchedLog(
-        this: unknown,
+        this: never,
         ...args: Parameters<typeof original>
       ) {
-        const span = trace.getSpan(context.active());
+        let record: Record<string, any> = {};
+        const config = instrumentation.getConfig();
+        if (!config.disableLogCorrelation) {
+          const span = trace.getSpan(context.active());
+          if (span) {
+            const spanContext = span.spanContext();
+            if (isSpanContextValid(spanContext)) {
+              record = injectRecord(spanContext, record);
+              instrumentation._callHook(span, record);
```

Comment:
There seems to be a bit of code duplication with `_getPatchedWrite`. Maybe it could be extracted into a helper?